### PR TITLE
feat(b-tabs): add props to not render aria-posinset and aria-setsize (closes #5054)

### LINKED ISSUE #5054: W3C validation error using b-tabs/b-tab
### Describe the bug

If I use th b-tabs/b-tab component for creating a tabulator, I get error messages when validating the generated code:
- Attribute aria-setsize not allowed on element a at this point.
- Attribute aria-posinset not allowed on element a at this point.

These messages are repeated for every tab.
I get the same error messages validating this url:
https://bootstrap-vue.org/docs/components/tabs

### Steps to reproduce the bug

1. Go to https://bootstrap-vue.org/docs/components/tabs
2. Use bookmarklet "Check serialized DOM of current page" (from: https://validator.w3.org/nu/about.html)
3. Use wcag-bookmarklet "Check for WCAG 2.0 parsing compliance" (from: https://validator.w3.org/nu/about.html) on the result list
4. You now see the errors which mark the page as invalid in a wcag test

### Expected behavior

No errors in the tab module are shown.

### Versions

**Libraries:**

- BootstrapVue: 2.1.0
- Bootstrap: 4.3.1
- Vue: 2.6.10

**Environment:**

any device, any browser

### PR DESCRIPTION
**Pull request by jenemde**
### Describe the PR

This PR offers the opportunity to prevent rendering of aria-posinset and aria-setsize in `<b-tabs>`.
A positive result from the Nu HTML validator is prerequisite for a wcag/bitv validation. The Nu HTML validator shows the errors

- Attribute aria-setsize not allowed on element a at this point.
- Attribute aria-posinset not allowed on element a at this point.

when validating the `<b-tabs>` component.

Not regarding if the beavior of the Nu HTML validator is wrong or not, the new properties `hidePosInSet` and `hideSetSize` offer the opportunity to prevent rendering of aria-posinset and aria-setsize in `<b-tabs>`. If these props are not set to true, the aria attributes are rendered.

[WAI-ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1) spec shows aria-posinset and aria-setsize as being valid on role="tab"

- https://www.w3.org/TR/wai-aria-1.1/#aria-posinset
- https://www.w3.org/TR/wai-aria-1.1/#aria-setsize

But [WAI-ARIA 1.1 best practices](https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html) does not use these attributes.

Closes https://github.com/bootstrap-vue/bootstrap-vue/issues/5054.

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix (fixes a boo-boo in the code) - `fix(...)`, requires a patch version update
- [x] Feature (adds a new feature to BootstrapVue) - `feat(...)`, requires a minor version update
- [ ] Enhancement (augments an existing feature) - `feat(...)`, requires a minor version update
- [ ] ARIA accessibility (fixes or improves ARIA accessibility) - `fix(...)`, requires a patch or minor version update
- [ ] Documentation update (improves documentation or typo fixes) - `chore(docs)`, requires a patch version update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe since breaking changes require a minor version update)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples`, `chore(docs): fix typo in README`, etc.). **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type (patch or minor).**

**If new features/enhancement/fixes are added or changed:**

- [x] Includes documentation updates
- [x] Includes component package.json meta section updates (prop, slot and event changes/updates)
- [ ] Includes any needed TypeScript declaration file updates
- [x] New/updated tests are included and passing (required for new features and enhancements)
- [x] Existing test suites are passing
- [x] CodeCov for patch has met target (all changes/updates have been tested)
- [x] The changes have not impacted the functionality of other components or directives
- [x] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

**Comment by jackmu95** [comment]
Thanks @jenemde for the every will put together PR!

The problem ist that these attributes are valid as of [WAI-ARIA `1.1`](https://www.w3.org/TR/wai-aria-1.1) as Troy already mentions in the issue:

> Strange, as the [WAI-ARIA `1.1`](https://www.w3.org/TR/wai-aria-1.1) spec shows `aria-posinset` and `aria-setsize` as being valid on `role="tab"`
> 
> * https://www.w3.org/TR/wai-aria-1.1/#aria-posinset
> * https://www.w3.org/TR/wai-aria-1.1/#aria-setsize

I don't see us to add props to remove/hide `aria-*` attrs.
We should rather fix the root of the issue itself - the validator.

Can you please report the the behavior at https://github.com/validator/validator.

**Comment by jenemde** [comment]
Thanks @jackmu95 for your very quick response.
I see the problem and I will add a comment/issue at https://github.com/validator/validator.

But the problem with the two aria attributes is that they are valid but not necessarily needed ("Not required if all elements in the set are present in the DOM"), so my PR wants to give the opportunity to dispense with them. The user has to do this explicitly so he is still responsible for the accessibility of the code. Maybe you could give a second thought to this.

Anyway, thanks a lot for your support. I will let you know about the reaction of https://github.com/validator/validator.

File: src/components/tabs/tabs.spec.js [hunk 1]
```diff
@@ -723,6 +723,43 @@ describe('tabs', () => {
     wrapper.destroy()
   })
 
+  it('does not render aria-posinset and aria-setsize if props to hide  are set to true', async () => {
```
Comment (DM2489) [inline src/components/tabs/tabs.spec.js:726]: whitespace between 'hide' and 'are'

File: src/components/tabs/tabs.js [hunk 2]
```diff
@@ -574,8 +578,8 @@ export const BTabs = /*#__PURE__*/ Vue.extend({
           controls: safeId ? safeId() : null,
           id: $tab.controlledBy || (safeId ? safeId(`_BV_tab_button_`) : null),
           noKeyNav,
-          posInSet: index + 1,
-          setSize: $tabs.length,
+          posInSet: hidePosinset ? null : index + 1,
```
Comment (DM2489) [inline src/components/tabs/tabs.js:581]: I'd suggest that the camelcase convention is followed - `hidePosInSet` and `hideSetSize`. These follow the camelcase convention that looks to be being used.

File: src/components/tabs/tabs.spec.js (hunk 1 above)
Comment (jenemde) [reply src/components/tabs/tabs.spec.js]: Thanks, whitespace is removed.

File: src/components/tabs/tabs.js (hunk 2 above)
Comment (jenemde) [reply src/components/tabs/tabs.js]: Thank you, I'v changed the prop names following the camelcase convention.

**Comment by jackmu95** [comment]
> The problem ist that these attributes are valid as of WAI-ARIA 1.1 as Troy already mentions in the issue:
> 
> Strange, as the WAI-ARIA 1.1 spec shows aria-posinset and aria-setsize as being valid on role="tab"
> 
> https://www.w3.org/TR/wai-aria-1.1/#aria-posinset
> https://www.w3.org/TR/wai-aria-1.1/#aria-setsize
>
> I don't see us to add props to remove/hide aria-* attrs.
> We should rather fix the root of the issue itself - the validator.

The related issue in the validator was finally fixed:
https://github.com/validator/validator/issues/1060

**Pull request closed without merging**